src/ldp/auxiliary/RdfValidator.ts

```diff
@@ -0,0 +1,42 @@
+import arrayifyStream from 'arrayify-stream';
+import type { RepresentationConverter } from '../../storage/conversion/RepresentationConverter';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { guardedStreamFrom } from '../../util/StreamUtil';
+import { BasicRepresentation } from '../representation/BasicRepresentation';
+import type { Representation } from '../representation/Representation';
+import { Validator } from './Validator';
+
+/**
+ * Validates a Representation by verifying if the data stream contains valid RDF data.
+ * It does this by letting the stored RepresentationConverter convert the data.
+ */
+export class RdfValidator extends Validator {
+  protected readonly converter: RepresentationConverter;
+
+  public constructor(converter: RepresentationConverter) {
+    super();
+    this.converter = converter;
+  }
+
+  public async handle(representation: Representation): Promise<void> {
+    // If the data already is quads format we know it's RDF
+    if (representation.metadata.contentType === INTERNAL_QUADS) {
+      return;
+    }
+
+    const identifier = { path: representation.metadata.identifier.value };
+
+    // Read data in memory first so it does not get lost
+    const data = await arrayifyStream(representation.data);
+    const preferences = { type: { [INTERNAL_QUADS]: 1 }};
+
+    // Creating new representation since converter might edit metadata
+    const tempRepresentation = new BasicRepresentation(data, identifier, representation.metadata.contentType);
+    const result = await this.converter.handleSafe({ identifier, representation: tempRepresentation, preferences });
+    // Drain stream to make sure data was parsed correctly
+    await arrayifyStream(result.data);
```

Comment:
In a later PR, let's create a utility for cloning a representation.
Perhaps mark this as TODO already.

But I'm missing a comment saying: we're just parsing the whole thing here.

I wonder whether we can double pipe (but I remember this causing a hanging issue at some point.

Reply:
Double piping is dangerous and can cause some weird errors if for example one of the pipes errors further down the line. Also backpressuring can get more complicated there. So I would prefer to avoid it as much as possible.

Reply:
Agreed on double piping; but the risk is memory hogging. I would strongly suggest to make this a utility method, so we can at least keep track of where it is used.